torch/distributions/relaxed_categorical.py

```diff
@@ -77,7 +77,7 @@ def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
         logits, value = broadcast_all(self.logits, value)
-        log_scale = (self.temperature.new(self.temperature.shape).fill_(K).lgamma() -
+        log_scale = (self.temperature.new_tensor(float(K)).lgamma().expand(self.temperature.shape) -
```

Comment:
`new_tensor` is a legacy constructor, so it would be nice to get rid of it. Since this looks like a scalar tensor, wouldn't it be enough to do `math.float(K) - ...` (just make lhs a Python scalar)

Reply:
Ha ha I'm glad you make the same mistakes I make :smile: